### PR TITLE
Issue 5368 - Retro Changelog trimming does not work

### PR DESCRIPTION
**Pull request by tbordaz**
Bug description:
	After an initial trimming, the trimming thread
	is not scheduled according to nsslapd-changelog-trim-interval.
	It is erroneously scheduled according to
        nsslapd-changelogmaxage.

Fix description:
	Get the trimming interval into 'ts' (trimming structure)
        and decide to trim according to nsslapd-changelogmaxage

relates: #5368

Reviewed by:

### CODE REVIEW

File: dirsrvtests/tests/suites/retrocl/basic_test.py [hunk 1]
```diff
@@ -329,6 +329,82 @@ def test_retrocl_trimming(topology_st):
...
+    assert inst.searchErrorsLog("trim_changelog: removed 9 change records")
+    for idx in range(2,11):
+        pattern="changenumber=%d,cn=changelog" % idx
+        assert inst.searchAccessLog(pattern)
+
+    # Clean up
+    inst.restart()
+    inst.config.set('nsslapd-errorlog-level', '0')
+    inst.config.set('nsslapd-plugin-logging', 'off')
+    inst.config.set('nsslapd-accesslog-level','256')
```
Comment (mreynolds389) [inline dirsrvtests/tests/suites/retrocl/basic_test.py]: Very minor, and not forcing you to do this, but perhaps this resetting of the config should be done as a finalizer function?

**Review by mreynolds389 — APPROVED** [review]
LGTM

**Comment by droideck** [comment]
The test `dirsrvtests/tests/suites/retrocl/basic_test.py::test_retrocl_trimming_interval` has failed (probably some minor test issue), rerunning it with the debug enabled.

```
        assert inst.searchErrorsLog("trim_changelog: removed 9 change records")
        for idx in range(2,11):
            pattern="changenumber=%d,cn=changelog" % idx
>           assert inst.searchAccessLog(pattern)
E           AssertionError: assert False
E            +  where False = <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>>('changenumber=2,cn=changelog')
E            +    where <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>> = <lib389.DirSrv object at 0x7fa6e5dff6a0>.searchAccessLog
```

**Comment by tbordaz** [comment]
> The test `dirsrvtests/tests/suites/retrocl/basic_test.py::test_retrocl_trimming_interval` has failed (probably some minor test issue), rerunning it with the debug enabled.
> 
> ```
>         assert inst.searchErrorsLog("trim_changelog: removed 9 change records")
>         for idx in range(2,11):
>             pattern="changenumber=%d,cn=changelog" % idx
> >           assert inst.searchAccessLog(pattern)
> E           AssertionError: assert False
> E            +  where False = <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>>('changenumber=2,cn=changelog')
> E            +    where <bound method DirSrv.searchAccessLog of <lib389.DirSrv object at 0x7fa6e5dff6a0>> = <lib389.DirSrv object at 0x7fa6e5dff6a0>.searchAccessLog
> ```

Yes it is failing on PRCI while it succeeds on my laptop :(
The (current) issue is that the numbering of the updates in retroCL is different on my laptop and on PRCI. ATM I am trying to run the test with a include scope, crossing fingers

**Comment by tbordaz** [comment]
I fixed the testcase. The problem is that for some unknown reason the changenumber attribute in the retroCL does not start with '1'. I modified the testcase to retrieve the starting number.

**Pull request merged**